libbrowser/src/libbrowser_wkwebview.mm

```diff
@@ -1418,21 +1520,70 @@ - (void)continueRequest:(MCWKWebViewBrowserNavigationRequest *)request withDecis
 	}
 	else
 	{
+		// continuing this request will cause any in-progress requests to be
+		// silently cancelled, so end that request here and send an appropriate
+		// error message.
 		if (m_main_request != nil)
-		{
-			// Redirect: replace current request with new one
-			m_main_request->Release();
-			m_main_request = nil;
-		}
+			[self finishNavigationWithError: "navigation request cancelled"];
+
```

Comment:
@livecodeian I thought this was a leak of `m_main_request` until I found it is released in `finishNavigationWithError`. Not sure if that could be made clearer in the method name.

Reply:
Yes I think there's a cleaner way to organise this - will see what I can do